packages/block-library/src/query/block.json

```diff
@@ -0,0 +1,45 @@
+{
+	"name": "core/query",
+	"category": "layout",
+	"attributes": {
+		"className": {
+			"type": "string"
+		},
+		"criteria": {
```

Comment:
After using this block for a few minutes, I found that there's a lot of work to be done to make sure the block works properly even without criteria/filtering...  

I was wondering what do you think if we split this PR into two, one block without filtering where we'd only focus on how the template gets edited/updated and rendered. and the filtering can be easily added on a follow-up PR when the interactions are sorted-out?